eyelid-match-ops/src/primitives/poly.rs

```diff
@@ -29,3 +31,89 @@ pub mod test;
 
 // Do not add code here.
 // Add functions or trait impls to modular_poly/*.rs and inherent method impls to modular_poly.rs.
+
+/// Returns the monic inverse of `a` in the cyclotomic ring, if it exists.
+/// Otherwise, returns an error.
+pub fn inverse<const MAX_POLY_DEGREE: usize>(
+    a: &Poly<MAX_POLY_DEGREE>,
+) -> Result<Poly<MAX_POLY_DEGREE>, String> {
+    let unreduced_mod_pol = Poly::new_unreduced_poly_modulus_slow();
+
+    let (_x, y, d) = extended_gcd(&unreduced_mod_pol, a);
+
+    // If `d` is a non-zero constant, we can compute the inverse of `d`,
+    // and calculate the final inverse.
+    if d.is_zero() {
+        Err("Can't invert the zero polynomial".to_string())
+    } else if d.degree() > 0 {
+        Err("Non-invertible polynomial".to_string())
+    } else {
+        // Reduce to a monic polynomial.
+        let mut inv: Poly<MAX_POLY_DEGREE> = y;
+        let divisor_inv: Coeff = d[0].inverse().expect("just checked for zero");
+
+        inv *= divisor_inv;
+
+        Ok(inv)
+    }
```

Comment:
Is `monic` the correct term here?
https://en.wikipedia.org/wiki/Polynomial_greatest_common_divisor#General_definition

If we want the inverse to be monic, should we be multiplying by the inverse of the coefficient of the degree?
That would be `y[y.degree()].inverse()`, instead of `d[0].inverse()`.

Reply:
@bgillesp this might be a maths question for you?

Reply:
Here we compute a primitive polynomial by dividing by the content. I fixed the terminology and added a reference to the description of the algorithm.